### PR TITLE
Wagtail 6

### PR DESCRIPTION
**Pull request by katdom13**
[Support Ticket](https://torchbox.monday.com/boards/1124794299/pulses/1396934613)
[Internal PR with Dan's initial comments](https://github.com/torchbox-forks/wagtail-footnotes/pull/2)

### Summary of changes

- Added Wagtail 6.0 compatibility
- Added tests for Django 5.0
- Added tests for Python 3.12

### CODE REVIEW

File: wagtail_footnotes/widgets.py [hunk 1]
```diff
@@ -1,20 +1,64 @@
...
+            try:
+                attrs["id"]
+            except (KeyError, TypeError) as exc:
+                raise TypeError(
+                    "ReadonlyUUIDInput cannot be rendered without an 'id' attribute"
+                ) from exc
+
+            widget_html = self.render_html(name, value, attrs)
+
+            return mark_safe(widget_html)  # noqa: S308
```
Comment (katdom13) [inline wagtail_footnotes/widgets.py:28]: This section was copied from Wagtail's `WidgetWithScript`:
https://github.com/wagtail/wagtail/blob/main/wagtail/utils/widgets.py#L27

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,8 +2,10 @@
 
 ## Unreleased
 
-- Add support for Wagtail 5.2
-- Add Wagtail 5.2 and Python 3.12 in test matrices @katdom13
+- Drop support for Wagtail < 5.2 @katdom13
+- Add support for Wagtail 6.0 @katdom13
+- Add support for Django 5.0 by @katdom13
+- Add compatibility tests for Wagtail 6.0 and Python 3.12 @katdom13
```
Comment (zerolab) [inline CHANGELOG.md:8]:
```suggestion
- Drop support for Wagtail < 5.2 @katdom13
- Add support for Wagtail 6.0/6.1, Django 5.0 and Python 3.12 @katdom13
```

**Review by zerolab — CHANGES_REQUESTED** [review]
It is shaping up well. A few tweaks needed

File: tox.ini [hunk 3]
```diff
@@ -28,14 +27,14 @@ deps =
...
 
     django3.2: Django>=3.2,<4.0
-    django4.1: Django>=4.1,<4.2
     django4.2: Django>=4.2,<4.3
+    django5.0: Django>=5.0,<5.1
 
-    wagtail4.1: wagtail>=4.1,<4.2
-    wagtail5.1: wagtail>=5.1,<5.2
     wagtail5.2: wagtail>=5.2,<5.3
+    wagtail6.0: wagtail>=6.0,<6.1
```
[the remark this answers is not in the document]
Comment (katdom13) [reply tox.ini:32]: Hi @zerolab ,
Updated here: https://github.com/torchbox/wagtail-footnotes/pull/66/commits/8219a7f7f0e290b72e80069315c13f7ef8e236ad

File: wagtail_footnotes/static/footnotes/js/read-only-uuid-controller.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// myapp/static/js/read-only-uuid-controller.js
+
```
[the remark this answers is not in the document]
Comment (katdom13) [reply wagtail_footnotes/static/footnotes/js/read-only-uuid-controller.js]: https://github.com/torchbox/wagtail-footnotes/pull/66/commits/b374a5afbf3f325be6bb1bf268794feb2bfcb681

File: wagtail_footnotes/widgets.py [hunk 5]
```diff
@@ -1,20 +1,64 @@
...
+from wagtail import VERSION as WAGTAIL_VERSION
 
 
-class ReadonlyUUIDInput(WidgetWithScript, HiddenInput):
-    """
-    This isn't really read-only. It's a hidden input with an an adjacent div
-    showing the current value; that way we can set the value in JavaScript, but
-    the user can't easily change it.
-    """
+if WAGTAIL_VERSION and WAGTAIL_VERSION >= (6, 0):
```
[the remark this answers is not in the document]
Comment (katdom13) [reply wagtail_footnotes/widgets.py]: https://github.com/torchbox/wagtail-footnotes/pull/66/commits/1f4f21b4af2a926e5d19314758a8ea143e0355f0

File: CHANGELOG.md (hunk 2 above)
Comment (katdom13) [reply CHANGELOG.md]: https://github.com/torchbox/wagtail-footnotes/pull/66/commits/3dbc60a51883cdd00eb1ea08ca7e92c0a2182638

**Comment by katdom13** [comment]
@zerolab ,
Could you please have a quick look at the latest changes?
Thank you!

File: tox.ini [hunk 6]
```diff
@@ -27,15 +25,15 @@ pass_env =
 deps =
     coverage[toml]>=7.2,<8.0
```
Comment (zerolab) [inline tox.ini:26]:
```suggestion
```

should not be needed since you added `extras = testing`

**Review by zerolab — APPROVED** [review]

**Comment by zerolab** [comment]
Thank you @katdom13

**Pull request merged**